### PR TITLE
Convert dashboard test into a system test

### PR DESCRIPTION
**Pull request by AntonKhorev**
Addresses https://github.com/openstreetmap/openstreetmap-website/pull/5070#issuecomment-2304501335 about `dashboards_controller_test.rb`. I converted the test into a system test because this way it's easier to test for the displayed content. I'm checking for "You have not added any friends" where previously was no assertion. Also I don't rely on a html element id that was used only for testing.

### CODE REVIEW

File: test/system/dashboard_test.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+require "application_system_test_case"
+
+class DashboardSystemTest < ApplicationSystemTestCase
```
Comment (tomhughes) [inline test/system/dashboard_test.rb:3]: Shouldn't this just be `DashboardTest` to match the filename?

Comment (AntonKhorev) [reply test/system/dashboard_test.rb:3]: I had problems with some system test previously that had the same name as some other test. I did the same thing as https://github.com/openstreetmap/openstreetmap-website/commit/45e6ebcafc5741750357b7fd6e11ec96c2cbfb24.

Comment (AntonKhorev) [reply test/system/dashboard_test.rb:3]: We have test named after:
- a thing in singular
- a thing in plural
- an action as a noun
- an action as a verb

In the [guides](https://guides.rubyonrails.org/v7.1/testing.html#system-testing) they have system tests named after things in plural and model tests after things in singular. If we want to do the same, we'll have to rename, for example, `ConfirmationResendSystemTest` to `ConfirmationsTest`.

**Comment by tomhughes** [comment]
Let's merge this now and we can worry about standardising test names later.

**Pull request merged**